openforcefield/topology/molecule.py

```diff
@@ -3288,8 +3292,8 @@ def to_iupac(self):
         from openeye import oeiupac
         return oeiupac.OECreateIUPACName(self.to_openeye())
 
-    @staticmethod
-    def from_topology(topology):
+    @classmethod
+    def from_topology(cls, topology):
```

Comment:
Reference molecules in a topology are (currently) always `FrozenMolecule`s, so this should remain as a staticmethod of the `FrozenMolecule` class.

Reply:
Actually, I take that back. We can always upcast it to whatever the user wants, as long as the constructor for the subclass can take a `FrozenMolecule` as input.